writer.go

```diff
@@ -99,27 +153,35 @@ func newWriterDictLevel(w io.Writer, cd *CDict, compressionLevel int) *Writer {
 }
 
 // Reset resets zw to write to w using the given dictionary cd and the given
-// compressionLevel.
+// compressionLevel. Other parameters set via WriterParams remain unchanged.
 func (zw *Writer) Reset(w io.Writer, cd *CDict, compressionLevel int) {
 	zw.inBuf.size = 0
 	zw.inBuf.pos = 0
 	zw.outBuf.size = cstreamOutBufSize
 	zw.outBuf.pos = 0
 
 	zw.cd = cd
-	initCStream(zw.cs, zw.cd, compressionLevel)
+	params := &WriterParams{
+		CompressionLevel: compressionLevel,
+		Dict:             cd,
+		WindowLog:        zw.wlog,
+	}
+	initCStream(zw.cs, params)
 
 	zw.w = w
 }
 
-func initCStream(cs *C.ZSTD_CStream, cd *CDict, compressionLevel int) {
-	if cd != nil {
-		result := C.ZSTD_initCStream_usingCDict(cs, cd.p)
+func initCStream(cs *C.ZSTD_CStream, params *WriterParams) {
+	if params.Dict != nil {
+		result := C.ZSTD_initCStream_usingCDict(cs, params.Dict.p)
 		ensureNoError("ZSTD_initCStream_usingCDict", result)
 	} else {
-		result := C.ZSTD_initCStream(cs, C.int(compressionLevel))
+		result := C.ZSTD_initCStream(cs, C.int(params.CompressionLevel))
 		ensureNoError("ZSTD_initCStream", result)
 	}
+
+	result := C.ZSTD_CCtx_setParameter(cs, C.ZSTD_cParameter(C.ZSTD_c_windowLog), C.int(params.WindowLog))
```

Comment:
Do we need adjusting `params.WindowLog` to default value when it is set to 0 or `zstd` properly handles zero value for this arg?

Reply:
`zstd` properly handles the zero value.

https://github.com/facebook/zstd/blob/5b0a452cac9672a3003781541235dc47727e6c2a/lib/zstd.h#L284-L291